test/loginjection.test.ts

```diff
@@ -54,32 +63,83 @@ describe('log injection', () => {
 
   describe('default flow', () => {
     before(() => {
-      startTracing({ logInjectionEnabled: true, serviceName: 'test-service' });
+      startTracing({ serviceName: 'test-service' });
     });
 
     after(() => {
       stopTracing();
     });
 
-    it('injects context to bunyan records', done => {
+    it('injects context to bunyan records', () => {
       const logger: bunyan = require('bunyan').createLogger({
         name: 'test',
         stream,
       });
-      assertInjection(logger, done);
+      assertInjection(logger);
     });
 
-    it('injects context to pino records', done => {
-      const logger: pino = require('pino')(stream);
-      assertInjection(logger, done);
+    it('injects context to pino records', () => {
+      const logger: pino.Logger = require('pino')(stream);
+      assertInjection(logger);
     });
 
-    it('injects context to winston records', done => {
+    it('injects context to winston records', () => {
       const winston: winston = require('winston');
       const logger = winston.createLogger({
         transports: [new winston.transports.Stream({ stream })],
       });
-      assertInjection(logger, done);
+      assertInjection(logger);
+    });
+  });
+
+  describe('injecting with custom hook', () => {
+    afterEach(() => {
+      stopTracing();
+    });
+
+    it('is possible to opt out from injecting resource attributes', () => {
+      const MY_VALUE = 'myValue';
```

Comment:
We have more than one test case using the same values, let's randomise or make these different for each test case.